src/middlewared/middlewared/validators.py

```diff
@@ -34,6 +35,25 @@ def __call__(self, value):
             raise ShouldBe("valid IP address")
 
 
+class Time:
+    def __call__(self, value):
+        try:
+            hours, minutes = value.split(':')
+        except ValueError:
+            raise ShouldBe('Time should be in 24 hour format like "18:00"')
+        else:
+            time_dict = {'hours': hours, 'minutes': minutes}
```

Comment:
I dont understand, why do you need time_dict? That seems to be complicating things? You dont see to use that for anything other than validation?

Providing wrong value to time() should raise TypeError either way.